engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -119,9 +114,9 @@ func (jm *JobMaster) InitImpl(ctx context.Context) error {
 		return err
 	}
 	if err := jm.preCheck(ctx, jm.initJobCfg); err != nil {
-		return err
+		return jm.Exit(ctx, frameModel.WorkerStatus{Code: frameModel.WorkerStatusError}, err)
```

Comment:
why we need to explicit call Exit, but in other branches we don't?

Reply:
If we return error in InitImpl, framwork will create a new jobmaster instead. If we don't expect that, we self exit the job.

Reply:
preCheck is the only non-retryable error in InitImpl. For other callback, we need to Sort it out